webpack/scenes/Hosts/ChangeContentSource/helpers.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { foremanUrl } from 'foremanReact/common/helpers';
+import { STATUS } from 'foremanReact/constants';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { Alert,
+  List,
+  ListItem,
+  Grid,
+  GridItem,
+  FormGroup,
+  FormSelect,
+  FormSelectOption } from '@patternfly/react-core';
+
+export const getHostIds = () => {
+  const cookie = document.cookie.split('; ')
+    .find(row => row.startsWith('_ForemanSelectedhosts'));
+  const params = new URLSearchParams(cookie);
+  const ids = params.get('_ForemanSelectedhosts');
+
+  if (ids) {
+    return (JSON.parse(ids));
+  }
+  return ([]);
+};
+
+export const formIsLoading = (data, contentView, change) => (
+  data === STATUS.PENDING ||
+  contentView === STATUS.PENDING ||
+  change === STATUS.PENDING);
+
+const invalidHostsList = hostsWithoutCs => (hostsWithoutCs.map(name => (
+  <List>
+    <ListItem>
+      <a href={foremanUrl(`/hosts/${name}`)}>{name}</a>
+    </ListItem>
+  </List>)));
+
+export const ignoredHostsAlert = hostsWithoutCs => (
+  <Grid>
+    <GridItem span={7}>
+      <Alert
+        variant="warning"
+        title={__('Some hosts are ignored!')}
+        className="cs_alert"
+        isExpandable
+      >
+        <p>
+          {__('Following hosts are not managed by Katello, so they will be ignored:')}
+          {invalidHostsList(hostsWithoutCs)}
+        </p>
+      </Alert>
+    </GridItem>
+  </Grid>);
+
+export const noHostsAlert = hostsWithoutCs => (
```

Comment:
A function that returns JSX is not really a helper; instead it should be written as a component. So you can do 
```suggestion
export const NoHostsAlert = ({hostsWithoutCs}) => (
...
```

This will transform it into a React component that takes one prop, `hostsWithoutCs`.

And then, in your JSX in `ChangeContentSourcePage.js` where you have 
```js
return noHostsAlert(hostsWithoutContent)
```
instead you can do
```js
return <NoHostsAlert hostsWithoutCs={hostsWithoutContent} />
```

The functions that return JSX should be moved out of `helpers.js`.  You can either put them in the same file as `ChangeContentSourcePage`, or in another file in the same directory.

This same comment applies to all 4 (I think?) functions in this file that return JSX. :)